src/upload-string-values.js

```diff
@@ -0,0 +1,71 @@
+// -*- mode: js; indent-tabs-mode: nil; js-basic-offset: 4 -*-
+//
+// This file is part of Thingpedia
+//
+// Copyright 2019 The Board of Trustees of the Leland Stanford Junior University
+//
+// Author: Silei Xu <silei@cs.stanford.edu>
+//
+// See LICENSE for details
+"use strict";
+
+const Tp = require('thingpedia');
+const fs = require('fs');
+const FormData = require('form-data');
+
+module.exports = {
+    initArgparse(subparsers) {
+        const parser = subparsers.addParser('upload-string-values', {
+            addHelp: true,
+            description: "Upload a string value dataset to Thingpedia."
+        });
+        parser.addArgument('--file', {
+            required: true,
+            help: "The .tsv file of the string values to upload."
+        });
+        parser.addArgument('--type-name', {
+            required: true,
+            help: "The id (type) of the string dataset."
+        });
+        parser.addArgument('--name', {
+            required: true,
+            help: "The name of the string dataset."
+        });
+        parser.addArgument('--license', {
+            required: false,
+            defaultValue: "public-domain",
+            help: "The license of the string dataset."
+        });
+        parser.addArgument('--preprocessed', {
+            nargs: 0,
+            action: 'storeTrue',
+            help: 'If the values are already tokenized.',
+            defaultValue: false
+        });
+    },
+
+    async createUpload(args) {
+        const fd = new FormData();
+        fd.append('upload', fs.createReadStream(args.file), { file: 'strings.tsv', contentType: 'text/csv;charset=utf8' });
```

Comment:
It's TSV, not CSV, so `text/tab-separated-values`

(not that it matters, but it might in the future if we start allowing both formats)